src/avfoundation.cs

```diff
@@ -10868,4 +10868,131 @@ interface AVContentProposal : NSCopying
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		IAVContentKeySessionDelegate Delegate { get; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; }
+
+		[NullAllowed, Export ("delegateQueue")]
+		DispatchQueue DelegateQueue { get; }
+
+		[NullAllowed, Export ("storageURL")]
+		NSUrl StorageURL { get; }
+
+		[Export ("keySystem")]
+		string KeySystem { get; }
+
+		[Export ("expire")]
+		void Expire ();
+
+		[NullAllowed, Export ("contentProtectionSessionIdentifier")]
+		NSData ContentProtectionSessionIdentifier { get; }
+
+		[Export ("processContentKeyRequestWithIdentifier:initializationData:options:")]
+		void ProcessContentKeyRequest ([NullAllowed] NSObject identifier, [NullAllowed] NSData initializationData, [NullAllowed] NSDictionary<NSString, NSObject> options);
+
+		[Export ("renewExpiringResponseDataForContentKeyRequest:")]
+		void RenewExpiringResponseData (AVContentKeyRequest contentKeyRequest);
+	}
+
+	[TV (10,2), Mac (10,12,4), iOS (10,3)]
+	[BaseType (typeof (NSObject))]
+	interface AVContentKeyRequest {
+		[TV (10, 2), Mac (10, 12, 4), iOS (10, 3)]
+		[Field ("AVContentKeyRequestProtocolVersionsKey")]
+		NSString ProtocolVersions { get; }
+
+		[Export ("status")]
+		AVContentKeyRequestStatus Status { get; }
+
+		[NullAllowed, Export ("error")]
+		NSError Error { get; }
+
+		[NullAllowed, Export ("identifier")]
+		NSObject Identifier { get; }
+
+		[NullAllowed, Export ("initializationData")]
+		NSData InitializationData { get; }
+
+		[Export ("canProvidePersistableContentKey")]
+		bool CanProvidePersistableContentKey { get; }
+
+		[Export ("makeStreamingContentKeyRequestDataForApp:contentIdentifier:options:completionHandler:")]
+		void MakeStreamingContentKeyRequestData (NSData appIdentifier, [NullAllowed] NSData contentIdentifier, [NullAllowed] NSDictionary<NSString, NSObject> options, Action<NSData, NSError> handler);
+
+		[Export ("processContentKeyResponse:")]
+		void ProcessResponse (AVContentKeyResponse keyResponse);
+
+		[Export ("processContentKeyResponseError:")]
+		void ProcessError (NSError error);
+
+		[Export ("respondByRequestingPersistableContentKeyRequest")]
+		void RespondRequest ();
+	}
+
+	[TV (10,2), Mac (10,12,4), iOS (10,3)]
+	[BaseType (typeof (AVContentKeyRequest))]
+	interface AVPersistableContentKeyRequest {
+		[Export ("persistableContentKeyFromKeyVendorResponse:options:error:")]
+		NSData GetPersistableContentKey (NSData keyVendorResponse, [NullAllowed] NSDictionary<NSString, NSObject> options, out NSError outError);
+
+		[Export ("renewsExpiringResponseData")]
+		bool RenewsExpiringResponseData { get; }
+	}
+
+	[TV (10,2), Mac (10,12,4), iOS (10,3)]
+	[BaseType (typeof (NSObject))]
+	interface AVContentKeyResponse {
+		[Static]
+		[Export ("contentKeyResponseWithFairPlayStreamingKeyResponseData:")]
+		AVContentKeyResponse Create (NSData keyResponseData);
+	}
+
+	[TV (10,2), Mac (10,12,4), iOS (10,3)]
+	[Protocol, BaseType (typeof (NSObject))]
```

Comment:
You can remove the `[BaseType]` attribute too, for this case I think the protocol interface is enough.